src/modules/CdmSourceList/components/List/ModalCreateEdit/container.js

```diff
@@ -59,11 +59,28 @@ ModalCreateEdit.propTypes = {
   resetDataSource: PropTypes.func.isRequired,
 };
 
+function validateForm(state) {
+  const formValues = getFormValues(form.createDataSource)(state);
+  const dbmsType = get(state, 'form.createDataSource.values.dbmsType');
+  let requiredFields = ['name', 'dbmsType', 'connectionString', 'cdmSchema'];
+  if (dbmsType === 'BIGQUERY') {
+    const dataSourceData = get(state, 'cdmSourceList.dataSource.queryResult.result', {}, 'Object');
+    const isEdit = dataSourceData && dataSourceData.id;
+    if (!isEdit) {
```

Comment:
What is the logic related to the `if`?

Reply:
if the action is not editing existing data source, the `keyfile` is required. While editing user can upload new keyfile to overwrite existing one.